Connection Functionality.

S3784866 opened the pull request:
Added Connection functionality
- DataStore creates a unconfigured serial port then uses the Serial Port selected to finish configuration and start connection.

S3784866 committed:
Started on ADL-21, basic functions are done.

S3784866 committed:
Cleaning up code

S3784866 committed:
Cleaning up code further for PR

S3784866 committed:
Final cleanup till PR

S3784866 committed:
Linting

erxncxte commented on Dashboard/Views/ConnectionView.axaml:
```diff
@@ -64,7 +64,7 @@
     <!-- Connect Button -->
-    <Button Grid.Row="2" Content="Connect" Margin="0,10,50,50" FontWeight="Bold" HorizontalAlignment="Left"/>
+    <Button Grid.Row="2" Content="Connect" Name="ConnectButton" Margin="0,10,50,50" FontWeight="Bold" Command="{Binding ConnectToSerialPortCommand}" CommandParameter="{Binding SelectedSerialPort}" IsEnabled="True" HorizontalAlignment="Left"/>
         </Grid>
```
I think there should also be a disconnect button. When testing, I've had it display 'Connected' on COM21 when in reality it isn't connected and I wanted to disconnect to try another port but I had to just kill the app and reboot again.

S3784866 replied:
I can implement a disconnect button.
Also for the above comments
1/2. The File connection and IP connection methods are not implemented, I can disable them and add a disclaimer that they are not implemented right now.
3. We are getting the CarID from the datalogger at the moment. Talking to the AV logger is a stretch goal from my understanding right now.
4. If you launched the app without the sim, you can connect to COM21 since its just a serial port, The connection icon doesn't update since the port is not sending anything back to trigger the RawDataUpdated but I can fix that up so it triggers on initial connection. Currently if a connection actually fails, the app will output the exception to the debug console right now and not change the connect buttons properties.

erxncxte commented on Dashboard/Connectors/IConnector.cs:
```diff
@@ -12,9 +12,12 @@ public interface IConnector
 
     /// <summary>
-    ///     Handles setting up the connector to the data source.
+    ///     Handles setting up the connector when the DataStore.
```
Please make this sentence make sense

erxncxte replied:
![image](https://github.com/user-attachments/assets/438f8f56-b341-45af-b639-11acc749d2e9)
Is this supposed to say from the DataStore?

erxncxte commented on Dashboard/Connectors/Serial/SerialConnector.cs:
```diff
@@ -23,11 +23,14 @@ public partial class SerialConnector(ISerialPort comPort) : IConnector
         // Set up the connection to the serial port
-        comPort.Configure("COM22", 115200);
-
+        comPort.Configure(portName, 115200);
```
Since you are setting the baud rate here, do we need a drop down option for that on the connection page as well? Do we need to confirm with Nick O that the data logger connects with 115200?

erxncxte replied:
Disregard, I asked Nick O and he said it only works with 115200, my bad!

erxncxte commented on Dashboard/Connectors/Serial/SerialConnector.cs:
```diff
@@ -74,7 +78,12 @@ private void ParseMessage(string message)
-            throw new InvalidOperationException("Invalid message format received", new Exception(message));
+        {
+            // Skips message if the format is invalid so the thread doesn't kill itself.
+            Console.WriteLine("Invalid message format recevied");
```
received*

erxncxte replied:
![image](https://github.com/user-attachments/assets/7514da50-beae-4644-8d28-eff24dfe9a9e)
This is still a typo in the new commit

erxncxte commented on Dashboard/Views/ConnectionView.axaml.cs:
```diff
@@ -35,7 +54,7 @@ private async void FileSelectionClicked(object sender, RoutedEventArgs args)
         }
```
I think it would be good to add some validation of file type here. We would only really accept a csv file I would imagine. If we aren't ready to allow file replays, maybe just add a message here saying 'feature coming soon' or something and disable the functionality.

S3784866 replied:
Can do.

erxncxte commented on Dashboard/Views/ConnectionView.axaml.cs:
```diff
@@ -35,7 +54,7 @@ private async void FileSelectionClicked(object sender, RoutedEventArgs args)
 
-    private void ConnectionTypeUpdated(object sender, RoutedEventArgs args)
+    private void ConnectionTypeUpdated(object? sender, RoutedEventArgs args)
     {
```
I think also when the connection type is changed, it should disconnect from any current connection and reset the connection button back to 'Connect'

S3784866 replied:
I'm not sure that would be the best way of handling this, at least right now since the other methods of connections are not implemented, I think leaving it as is but adding the disconnect button and disabling the other connection types for now would be enough.

erxncxte reviewed (changes requested):
I also had some weird bugs when I tested this branch:
- When I clicked Connect with nothing entered in the IP address box it just said connected. I think there should be some validation here or disable the feature if it isn't going to be implemented in this project. Maybe show an error message to say you need to enter something.
- Same as above but with no file chosen for the File Connection type
- Are we getting Car ID from the datalogger? I thought you needed to enter it on the connection page.
- When I clicked COM21 as the port to connect to, it said connected yet nothing displayed in the console. I think if it is unsuccessful, it should show an error instead of falsely displaying connected.

S3784866 committed:
Addressed PR Comments:
- Implemented Disconnect button.
- Fixed connection icon not updating if there is not data coming from the serial port.
- Fixed comments

S3784866 committed:
Linting

erxncxte commented:
Once the last two comments regarding spelling are addressed, happy for this branch to be merged

S3784866 committed:
Fix comment typos
